### PR TITLE
Update persisted-model.js

### PR DESCRIPTION
**Pull request by crandmck**
Correct doc for findOrCreate per https://github.com/strongloop/loopback-datasource-juggler/issues/766

### CODE REVIEW

**Comment by crandmck** [comment]
@superkhau Please review and land if it looks OK.

File: lib/persisted-model.js [hunk 1]
```diff
@@ -110,19 +110,37 @@ module.exports = function(registry) {
     throwNotAttached(this.modelName, 'upsert');
   };
 
-  /**
-   * Find one record matching the optional `where` filter. The same as `find`, but limited to one object.
+/**
```
Comment (superkhau) [inline lib/persisted-model.js]: add two spaces in front

File: lib/persisted-model.js [hunk 2]
```diff
@@ -110,19 +110,37 @@ module.exports = function(registry) {
...
+   * { where: { key: val, key2: {gt: 'val2'}, ...} }
+   * ```
+   * <br/>See
+   * [Where filter](https://docs.strongloop.com/display/LB/Where+filter#Wherefilter-Whereclauseforqueries).
    * @param {Object} data Data to insert if object matching the `where` filter is not found.
    * @callback {Function} callback Callback function called with `cb(err, instance)` arguments.  Required.
    * @param {Error} err Error object; see [Error object](http://docs.strongloop.com/display/LB/Error+object).
    * @param {Object} instance Model instance matching the `where` filter, if found.
-   */
+*/
```
Comment (superkhau) [inline lib/persisted-model.js]: add two spaces in front

**Comment by superkhau** [comment]
Just missing tabs in front of the start and end of comment block. Otherwise, LGTM.

File: lib/persisted-model.js [hunk 3]
```diff
@@ -110,19 +110,37 @@ module.exports = function(registry) {
     throwNotAttached(this.modelName, 'upsert');
   };
 
-  /**
-   * Find one record matching the optional `where` filter. The same as `find`, but limited to one object.
+/**
+   * Finds one record matching the optional filter object. If not found, creates the object using
+   * the data provided as second argument.
```
Comment (superkhau) [inline lib/persisted-model.js]: data provided in the second argument?

File: lib/persisted-model.js [hunk 4]
```diff
@@ -110,19 +110,37 @@ module.exports = function(registry) {
...
+   * In this sense it it the same as `find`, but limited to one object.
```
Comment (superkhau) [inline lib/persisted-model.js]: it it -> it is

File: lib/persisted-model.js [hunk 5]
```diff
@@ -110,19 +110,37 @@ module.exports = function(registry) {
...
+   * <br/>See [Include filter](http://docs.strongloop.com/display/LB/Include+filter).
+   * @property {Number} limit Maximum number of instances to return.
+   * <br/>See [Limit filter](http://docs.strongloop.com/display/LB/Limit+filter).
+   * @property {String} order Sort order: either "ASC" for ascending or "DESC" for descending.
+   * <br/>See [Order filter](http://docs.strongloop.com/display/LB/Order+filter).
+   * @property {Number} skip Number of results to skip.
+   * <br/>See [Skip filter](http://docs.strongloop.com/display/LB/Skip+filter).
+   * @property {Object} where Where clause, like
+   * ```
+   * { where: { key: val, key2: {gt: 'val2'}, ...} }
```
Comment (superkhau) [inline lib/persisted-model.js]: {where: {key: 'val', key2: {gt: 'val2'}, ...}}

**Comment by crandmck** [comment]
Fixed, ty.

**Comment by crandmck** [comment]
OK, fixed the notional `where` clause.

File: lib/persisted-model.js [hunk 6]
```diff
@@ -111,13 +111,31 @@ module.exports = function(registry) {
   };
 
   /**
-   * Find one record matching the optional `where` filter. The same as `find`, but limited to one object.
+   * Finds one record matching the optional filter object. If not found, creates the object using
+   * the data provided as second argument.
+   * In this sense it it the same as `find`, but limited to one object.
```
Comment (superkhau) [inline lib/persisted-model.js]: it is

File: lib/persisted-model.js [hunk 7]
```diff
@@ -111,13 +111,31 @@ module.exports = function(registry) {
   };
 
   /**
-   * Find one record matching the optional `where` filter. The same as `find`, but limited to one object.
+   * Finds one record matching the optional filter object. If not found, creates the object using
+   * the data provided as second argument.
```
Comment (superkhau) [inline lib/persisted-model.js]: data provided in the second argument

File: lib/persisted-model.js [hunk 8]
```diff
@@ -111,13 +111,31 @@ module.exports = function(registry) {
...
+   * <br/>See [Include filter](http://docs.strongloop.com/display/LB/Include+filter).
+   * @property {Number} limit Maximum number of instances to return.
+   * <br/>See [Limit filter](http://docs.strongloop.com/display/LB/Limit+filter).
+   * @property {String} order Sort order: either "ASC" for ascending or "DESC" for descending.
+   * <br/>See [Order filter](http://docs.strongloop.com/display/LB/Order+filter).
+   * @property {Number} skip Number of results to skip.
+   * <br/>See [Skip filter](http://docs.strongloop.com/display/LB/Skip+filter).
+   * @property {Object} where Where clause, like
+   * ```
+   * { where: { key: val, key2: {gt: val2}, ...} }
```
Comment (superkhau) [inline lib/persisted-model.js]: where still has spaces around it

Comment (superkhau) [inline lib/persisted-model.js]: should look like:
`{where: {key: val, key2: {gt: val2}, ...}}`

**Comment by crandmck** [comment]
Thanks @superkhau !!!

**Pull request merged**